frontend/src/components/Pong2D.vue

```diff
@@ -0,0 +1,200 @@
+<template>
+  <div>
+    <canvas class="boards" ref="gamecanvas"></canvas>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, onUnmounted } from 'vue'
+import { type Rectangle, Paddle, type Circle, Ball } from '../types'
+
+const emit = defineEmits(['gameOver'])
+
+const gamecanvas = ref<HTMLCanvasElement | null>(null)
+let ctx = ref<CanvasRenderingContext2D | null>(null)
+let animation: number | null = null
+let ball: Circle | null = null
+let paddle1: Rectangle | null = null
+let paddle2: Rectangle | null = null
+let lastAnimationTime: DOMHighResTimeStamp | null = null
+let lastPaddleCollisionTime = 0
+let lastWallCollisionTime = 0
+
+onMounted(() => {
+  console.log('Mounted Pong')
+  window.addEventListener('resize', onWidthChange)
```

Comment:
how would the travel time on a smaller screen affect multiplayer gameplay?

Reply:
I used this event as a way to resize the canvas.

At the moment it resets the position of the ball and paddles, which is not great at all but otherwise it would overflow the paddles and ball.

I may need to set the position as a percentage of the total canvas x and y to allow the same experience on different size of board.
With socket.io I will try to play with 2 clients with boards with different sizes to check if it it works.

Reply:
I recommend having an aspect ratio for easy upscale or downscale on client side, while in the server it remains the same canvas

Reply:
But I don’t know how good that would work, only testing

Reply:
Ah, okay. So on the server side it would send a value like 400, 200 and then each client would take that value and do a ratio based on their canvas size. Is that it?

Reply:
yes something like that seems like it would work